python_backend/api.py

```
@@ -36,14 +182,268 @@ async def root():
             "health": "/health",
+            "extract-text": "/extract-text (POST)",
+            "apply-translations": "/apply-translations (POST)",
             "generate": "/generate (POST)",
+            "generate-previews": "/generate-previews (POST)",
         }
     }
 
 
+@app.post("/extract-text")
+async def extract_text_from_pptx(
+    file: UploadFile = File(...),
```

Ruff (0.14.10)

Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)